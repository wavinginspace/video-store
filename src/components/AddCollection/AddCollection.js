import React, { Component } from 'react';
import ApiContext from '../../ApiContext';
import config from '../../config';
import { Link } from 'react-router-dom';
import './AddCollection.scss';

class AddCollection extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: '',
      notes: '',
      fieldTouched: false
    };
  }

  static contextType = ApiContext;

  updateField(field, value) {
    let fieldTouched = `${field}Touched`;
    this.setState({
      [field]: value,
      [fieldTouched]: true
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { title, notes } = this.state;

    const data = {
      title,
      notes
    };

    fetch(`${config.API_ENDPOINT}/api/collections`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(data)
    })
      .then(res => {
        if (!res.ok) {
          throw new Error('there was an error');
        }
        return res.json();
      })
      .then(data => {
        this.context.addCollection(data);
        this.props.history.push('/');
      })
      .catch(err => {});
  }

  render() {
    return (
      <div className="AddCollection fadeIn">
        <form className="box" onSubmit={e => this.handleSubmit(e)}>
          <label htmlFor="title">Title:</label>
          <input type="text" name="title" autoComplete="off" onChange={e => this.updateField('title', e.target.value)}/>
          <label htmlFor="notes">Notes:</label>
          <textarea
            name="notes"
            id="collection-notes"
            cols="30"
            rows="10"
            onChange={e => this.updateField('notes', e.target.value)}></textarea>
          <button type="submit" className="submit-button">
            Submit
          </button>
        </form>
        <Link className="back-button" to="/">
          <p className="back-button-p">Back</p>
        </Link>
      </div>
    );
  }
}

export default AddCollection;
