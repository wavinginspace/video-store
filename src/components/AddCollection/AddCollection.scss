.add-collection.button {
  background-color:rgb(10, 254, 10);
}