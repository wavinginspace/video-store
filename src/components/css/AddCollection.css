:root {
  --main: var(--main);
  --nav: #499acc;
  --headergrad: linear-gradient(0deg, #ff7676, #fffd8f);
  --light: #fffd8f;
  --headerog: #ff7676;
}

.add-collection.button {
  background-color: #0afe0a;
}

.fadeIn {
  -webkit-animation-name: fadeIn;
          animation-name: fadeIn;
  -webkit-animation-duration: .2s;
          animation-duration: .2s;
  -webkit-animation-fill-mode: forwards;
          animation-fill-mode: forwards;
}

@-webkit-keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
/*# sourceMappingURL=AddCollection.css.map */