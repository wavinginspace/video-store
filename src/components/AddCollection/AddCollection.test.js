import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router} from 'react-router-dom';
import AddCollection from './AddCollection';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Router><AddCollection /></Router>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<Router><AddCollection /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
