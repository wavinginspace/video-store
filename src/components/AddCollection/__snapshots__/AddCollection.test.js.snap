// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="AddCollection fadeIn"
>
  <form
    className="box"
    onSubmit={[Function]}
  >
    <label
      htmlFor="title"
    >
      Title:
    </label>
    <input
      autoComplete="off"
      name="title"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="notes"
    >
      Notes:
    </label>
    <textarea
      cols="30"
      id="collection-notes"
      name="notes"
      onChange={[Function]}
      rows="10"
    />
    <button
      className="submit-button"
      type="submit"
    >
      Submit
    </button>
  </form>
  <a
    className="back-button"
    href="/"
    onClick={[Function]}
  >
    <p
      className="back-button-p"
    >
      Back
    </p>
  </a>
</div>
`;
